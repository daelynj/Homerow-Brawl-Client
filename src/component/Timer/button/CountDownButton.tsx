import * as React from "react";

interface Props {
  handleEvent: (newEvent: any) => void;
}

export const CountDownButton = (props: Props) => (
  <div>
    <button onClick={props.handleEvent}>Start Game</button>
  </div>
);
