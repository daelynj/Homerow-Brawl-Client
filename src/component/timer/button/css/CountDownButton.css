.start {
  margin-left: auto;
  margin-right: auto;
  width: 20%;
  text-align: center;
}
