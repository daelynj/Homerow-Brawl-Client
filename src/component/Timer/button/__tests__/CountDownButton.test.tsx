import * as React from "react";
import { CountDownButton } from "../CountDownButton";
import renderer from "react-test-renderer";
import { mount, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("CountDownButton", () => {
  describe("The button render", () => {
    const mockHandleEvent = jest.fn();

    it("renders a Start game button", () => {
      const tree = renderer
        .create(<CountDownButton handleEvent={mockHandleEvent} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("onClick", () => {
    const mockHandleEvent = jest.fn();

    it("triggers an event", () => {
      const wrapper = mount(<CountDownButton handleEvent={mockHandleEvent} />);

      wrapper.find("button").simulate("click");
      expect(mockHandleEvent).toHaveBeenCalled();
    });
  });
});
