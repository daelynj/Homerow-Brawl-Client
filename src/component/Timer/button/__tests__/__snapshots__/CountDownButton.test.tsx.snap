// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountDownButton The button render renders a Start game button 1`] = `
<div>
  <button
    onClick={[MockFunction]}
  >
    Start Game
  </button>
</div>
`;
